include/eve/module/core/regular/impl/any.hpp

```diff
@@ -47,23 +49,27 @@ any_(EVE_SUPPORTS(cpu_), C const& cond, T const& v) noexcept
   }
 }
 
+template<callable_options O>
 EVE_FORCEINLINE bool
-any_(EVE_SUPPORTS(cpu_), bool v) noexcept
+any_(EVE_REQUIRES(cpu_), O const&, bool v) noexcept
+requires match_option<condition_key, O, ignore_none_>
 {
   return v;
 }
 
-template<value T>
+template<callable_options O, value T>
 EVE_FORCEINLINE bool
-any_(EVE_SUPPORTS(cpu_), logical<T> const& v) noexcept
+any_(EVE_REQUIRES(cpu_), O const&, logical<T> const& v) noexcept
+requires match_option<condition_key, O, ignore_none_>
 {
   if constexpr( scalar_value<T> ) return bool(v);
-  else return eve::any[ignore_none](v);
+  else return eve::any(eve::top_bits { v });
 }
```

Comment:
This is a very weird overload.
I suggest to change above `bool` to `scalar_value or bool` and remove this overload